src/hooks/useAsync.js

```diff
@@ -0,0 +1,23 @@
+import { useState } from 'react';
+
+function useAsync(asyncFunction) {
+  const [pending, setPending] = useState(false);
+  const [error, setError] = useState(null);
+
+  const wrappedFunction = async (...args) => {
+    try {
+      setError(null);
+      setPending(true);
+      return await asyncFunction(...args);
+    } catch (error) {
+      setError(error);
+      return;
+    } finally {
+      setPending(false);
+    }
+  };
+
+  return [pending, error, wrappedFunction];
```

Comment:
이 커스텀훅을 사용하는 곳에서 
`const [, , getUserFolder] = useAsync(getMockFolder);` 와 같이 3번째 함수만 주로 사용하고 1, 2번째는 안쓰고 있는데요 
그렇다면 배열로 리턴하는 것보다 객체로 리턴하는 것이 더 편할 것 같아요 

```suggestion
  return { pending, error, wrappedFunction };
```

사용처에서는 사용할 값만 받아서 필요 시 이름 바꿔서 쓰기
```
const { wrappedFunction: getUserFolder } = useAsync(getMockFolder);
```

코드 변경하고 보니 useAsync 에서 리턴할 때 wrapperFunction 보다 좀 더 함수의 기능을 담은 네이밍으로 변경해서 내보내는 게 좋을 것 같군요!